src/operator/make_loss-inl.h

```diff
@@ -22,13 +22,29 @@ namespace op {
 namespace make_loss_enum {
 enum MakeLossOpInputs {kData};
 enum MakeLossOpOutputs {kOut};
+enum MakeLossOpType {kNull, kBatch, kValid};
+enum MakeLossOpResource {kTempSpace};
 }  // namespace make_loss_enum
 
 struct MakeLossParam : public dmlc::Parameter<MakeLossParam> {
   float grad_scale;
+  int normalization;
+  float valid_thresh;
   DMLC_DECLARE_PARAMETER(MakeLossParam) {
     DMLC_DECLARE_FIELD(grad_scale).set_default(1.0f)
     .describe("gradient scale as a supplement to unary and binary operators");
+    DMLC_DECLARE_FIELD(valid_thresh).set_default(0.0f)
```

Comment:
In a lot of case, we need to ignore the gradient from some elements because they are neither positive or negative or simply because we don't care. In practice we can set those elements to zero or negative before passing into a make_loss layer. This threshold sentinel the valid elements. 
Finally the gradients are normalized by the number of valid elements in valid normalization mode.